### PR TITLE
[IIIF-878] Create V3ManifestVerticle

### PR DESCRIPTION
**Pull request by markmatney**
Changes:
- bump JP 3.0 version
- change scope of jsoup dependency (depended on by JP 3.0)
- make CsvParser.getObjectType(s) static methods so they can be easily used in V3ManifestVerticle
- rename V3ManifestLabelComparator to V3CollectionItemLabelComparator
- create V3ManifestVerticle
- add v3 manifest test fixtures
- parameterize FakeS3BucketVerticle so that it can use either API version

### CODE REVIEW

**Comment by markmatney** [comment]
Also: I've tested manually with a few collections from the Eureka repo: `done/hathaway`, `done/ethiopic_mss`, and `done/ammenpapers`, and (1) compared the resulting manifests to the corresponding v2 ones on our production server and (2) viewed them in Mirador 3, which all looked good to me.

File: src/test/java/edu/ucla/library/iiif/fester/verticles/V3ManifestVerticleTest.java [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+/**
+ * Tests run against the ManifestVerticle.
+ */
+@RunWith(VertxUnitRunner.class)
+public class V3ManifestVerticleTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(V3ManifestVerticleTest.class, Constants.MESSAGES);
+
+    private static final String[] TEST_VERTICLES = new String[] { V3ManifestVerticle.class.getName(),
+        ManifestVerticle.class.getName(), FakeS3BucketVerticle.class.getName() };
```
Comment (ksclarke) [inline src/test/java/edu/ucla/library/iiif/fester/verticles/V3ManifestVerticleTest.java:63]: Sigh. One day FakeS3BucketVerticle will be refactored away. Looking forward to that day.

File: src/test/java/edu/ucla/library/iiif/fester/verticles/V3ManifestVerticleTest.java [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+                        }
+                    } else {
+
+                    }
+                });
+
+                myRunID = UUID.randomUUID().toString();
+            } else {
+                aContext.fail(getConfig.cause());
+                asyncTask.complete();
```
Comment (ksclarke) [inline src/test/java/edu/ucla/library/iiif/fester/verticles/V3ManifestVerticleTest.java:146]: We've not been consistent on this (figuring this is probably a copy and paste), but when you have a guaranteed failure you don't need to complete the task, the failure completes it.

File: src/test/java/edu/ucla/library/iiif/fester/verticles/V3ManifestVerticleTest.java [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+                                myJsonFiles = getS3TempDir(map.get(deploymentKey));
+                                asyncTask.complete();
+                            } catch (final FileNotFoundException details) {
+                                aContext.fail(details);
+                            }
+                        } else {
+                            aContext.fail(LOGGER.getMessage(MessageCodes.MFS_077, deploymentKey));
+                        }
+                    } else {
+
```
Comment (ksclarke) [inline src/test/java/edu/ucla/library/iiif/fester/verticles/V3ManifestVerticleTest.java:139]: An empty else?

File: src/main/java/edu/ucla/library/iiif/fester/verticles/S3BucketVerticle.java [hunk 4]
```diff
@@ -84,13 +84,14 @@ public void start() throws Exception {
         getJsonConsumer().handler(message -> {
             final JsonObject messageBody = message.body();
             final String action = message.headers().get(Constants.ACTION);
+            final String apiVersion = message.headers().get(Constants.IIIF_API_VERSION);
```
Comment (ksclarke) [inline src/main/java/edu/ucla/library/iiif/fester/verticles/S3BucketVerticle.java:87]: This is just in prep for when we treat these differently? Just logging the difference now?

**Review by ksclarke — COMMENTED** [review]
Looks good. Just one comment really about an empty else block.

File: src/main/java/edu/ucla/library/iiif/fester/verticles/S3BucketVerticle.java (hunk 4 above)
Comment (markmatney) [reply src/main/java/edu/ucla/library/iiif/fester/verticles/S3BucketVerticle.java:87]: I just thought it was something good to log

File: src/test/java/edu/ucla/library/iiif/fester/verticles/V3ManifestVerticleTest.java (hunk 3 above)
Comment (markmatney) [reply src/test/java/edu/ucla/library/iiif/fester/verticles/V3ManifestVerticleTest.java:139]: Looks like an issue propagated from ManifestVerticleTest from which I copied this test. I can add something here in both test classes.

**Review by ksclarke — APPROVED** [review]
:+1:

**Review by DRickard — APPROVED** [review]

**Pull request merged**